pkg/util/vmopv1/vm.go

```diff
@@ -398,13 +398,75 @@ func CnsRegisterVolumeToVirtualMachineMapper(
 	ctx context.Context,
 	k8sClient client.Client) handler.MapFunc {
 
-	// TODO(AllDisksArePVCs)
-	//
-	// Implement this function. Please see the function
-	// reconcileUnmanagedToManagedDisks from vmprovider_vm.go for more
-	// information on how to implement this function.
+	if ctx == nil {
+		panic("context is nil")
+	}
+	if k8sClient == nil {
+		panic("k8sClient is nil")
+	}
+
+	// For a given CnsRegisterVolume, return reconcile requests for the VM
+	// that owns it.
+	return func(ctx context.Context, o client.Object) []reconcile.Request {
+		if ctx == nil {
+			panic("context is nil")
+		}
+		if o == nil {
+			panic("object is nil")
+		}
+
+		logger := pkglog.FromContextOrDefault(ctx).
+			WithValues("name", o.GetName(), "namespace", o.GetNamespace())
+		logger.V(4).Info("Reconciling VMs due to CnsRegisterVolume event")
+
+		var requests []reconcile.Request
+
+		// Method 1: Check owner references for direct VM ownership
+		for _, ownerRef := range o.GetOwnerReferences() {
+			if ownerRef.Kind == "VirtualMachine" && ownerRef.APIVersion == vmopv1.GroupVersion.String() {
```

Comment:
After a schema bump, are we worried about skipping things are (still) owned by the prior version? (or there is something I haven't gotten to and we'll always update the OwnerRef)